src/main/java/org/alfresco/repo/site/SiteServiceImpl.java

```diff
@@ -1493,7 +1491,29 @@ public NodeRef doWork() throws Exception
             return siteNodeRef;
         }
     }
-    
+
+    private NodeRef getArchivedSiteNodeRef(String shortName)
+    {
+        final NodeRef siteRoot = getSiteParent(shortName);
+
+        return AuthenticationUtil.runAs(() -> {
+            NodeRef storeArchiveNode = directNodeService
+                .getStoreArchiveNode(siteRoot.getStoreRef());
+
+            List<ChildAssociationRef> childAssociationRefs = directNodeService
+                .getChildAssocs(storeArchiveNode, ContentModel.ASSOC_CHILDREN,
+                    NodeArchiveService.QNAME_ARCHIVED_ITEM);
+
+            Optional<NodeRef> archivedShortNameSite = childAssociationRefs.stream()
+                .map(childAssociationRef -> childAssociationRef.getChildRef())
+                .filter(site -> ((String) directNodeService.getProperty(site, ContentModel.PROP_NAME))
+                    .equalsIgnoreCase(shortName))
+                .findFirst();

```

Comment:
We should try to seek a different approach to find if the node is already archived. In the current approach, if we have a large amount of items in the trashcan then it will retrieve and iterate on all...